library/src/main/java/com/bumptech/glide/RequestManager.java

```diff
@@ -354,12 +367,19 @@ public synchronized void onStart() {
 
   /**
    * Lifecycle callback that unregisters for connectivity events (if the
-   * android.permission.ACCESS_NETWORK_STATE permission is present) and pauses in progress loads.
+   * android.permission.ACCESS_NETWORK_STATE permission is present) and pauses in progress loads
+   * and clears all resources if {@link #clearOnStop()} is called.
    */
   @Override
   public synchronized void onStop() {
     pauseRequests();
     targetTracker.onStop();
+    if (clearOnStop) {
```

Comment:
Could you extract a clearRequests() method and re-use it here and in onDestroy?

Reply:
Oh and I think you could then do:
```java
if (clearOnStop) {
  clearRequests();
} else { 
  pauseRequests();
}
```

Reply:
@sjudd thanks for the suggestion mate.
Please review, I have implemented.